server/api/talks.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express')
+const { catchErrors } = require('../helpers')
+const { Talk } = require('../models')
+const router = express.Router()
+
+module.exports = router
+
+const fetchTalks = async (req, res) => {
+    const talks = await Talk.find()
+    const data = {
+        talks,
+        total: talks.length,
+    }
+    res.json(data)
+}
+
+const createTalk = async (req, res) => {
+    const data = {
+        ...req.body,
+        vote: 0,
+    }
+    try {
+        const talk = await (new Talk(data)).save()
+        res.json(talk)
+    } catch (_) {
+        res.status(403).end()
+    }
+}
+
+const upvoteTalk = async (req, res) => {
+    const talk = await Talk.findOneAndUpdate(
+        { _id: req.params.id },
+        { $inc: { vote: 1 } },
+        { new: true }
+    )
+    res.json(talk)
+}
+
+
+router.get('/', catchErrors(fetchTalks))
+router.post('/', catchErrors(createTalk))
+router.put('/:id/upvote', catchErrors(upvoteTalk))
```

Comment:
Why do you prefer HTTP `put` instead of `post` here?

Reply:
As I know, `POST` is mainly used for create, while `PUT` is mainly for update.
On more thing I wanna mention is that I think the `patch` method is more suitable for this partial update scenario.

Reply:
Agreed `patch` would be better here, but here the issue is `put`should be idempotent, see(https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html 9.1.2) which is not good for your upvote logic.
> POST is mainly used for create, while PUT is mainly for update.

☝️  That's not true, same reason as mentioned above.

Reply:
Got what u mean and agreed with u. 🙌
I have another idea of designing this API - would it be better to use the `:/id` endpoint instead and write some logic to separate the update of the other fields, the upvote/downvote logic via parsing the http request body info here?  This is how I thought this API should be designed initially, which should be more scalable. The reason why I didn't do it this way is just because the other features are not required for the user story. Normally I don't over architect anything - yep, that may be an excuse of my laziness, lol.😅